src/classes/WalletConnectors/BrowserConnector.ts

```diff
@@ -0,0 +1,26 @@
+import { type Eip1193Provider } from 'ethers'
+import { WalletConnector } from './WalletConnector'
+
+export interface WindowProvider extends Eip1193Provider {
+    isMetaMask?: boolean
+    isTrust?: boolean
+    isSafePal?: boolean
+    isBitKeep?: boolean
+    providers?: WindowProvider[]
+}
+
+export class BrowserConnector extends WalletConnector<WindowProvider | undefined> {
+    public readonly id: string = 'browserConnector'
+    public readonly name: string = 'Browser Wallet'
+
+    public getProvider() {
+        if (typeof window === 'undefined') return
+
+        const ethereum = (window as unknown as { ethereum?: WindowProvider })
+            .ethereum
```

Comment:
since you're using `(window as unknown as { ethereum?: WindowProvider })` in all connectors that extend the BrowserConnector, it would be nice to have an internal function that returns a dynamically typed window object

Reply:
What about declaring a global `.d.ts` types file?

```ts
declare global {
    interface Window {
        ethereum?: WindowProvider
        trustwallet?: WindowProvider
        BinanceChain?: WindowProvider
        bitkeep?: { ethereum: WindowProvider }
    }
}
```